### ISSUE TITLE
Unformatted comment spans add extra newline

### ISSUE THREAD

**Issue by eureka-cpu** [FuelLabs/sway#2691]
```rust
script;

// use statements
use std::*;
```
formats into:
```rust
script;


// use statements
use std::*;
```
Each additional format call adds more extra newlines
2nd format:
```rust
script;



// use statements
use std::*;
```

**Comment by eureka-cpu** [comment]
Ref #2669

**Comment by mitchmindtree** [comment]
Re-opening as #2692 doesn't seem to have quite fixed the issues with the examples just yet.

Specifically, a few of the examples are still having issues with extra whitespace being inserted on each successive re-format. As mentioned in @eureka-cpu's OP, this *only* appears to occur if the whitespace is followed by a comment. cc @kayagokalp

**Comment by eureka-cpu** [comment]
```rust
script;

use std::assert::assert;

struct Foo {
    f1: u32,
    f2: b256,
}

fn main() {
    // Array of integers with type ascription
    let array_of_integers: [u8;
    5] = [1, 2, 3, 4, 5];

    // Array of strings
    let array_of_strings = [ "Bob", "Jan", "Ron", ];

    // Array of structs
    let array_of_structs: [Foo;
    2] = [Foo {
        f1: 11, f2: 0x1111111111111111111111111111111111111111111111111111111111111111, 
    },
    Foo {
        f1: 22, f2: 0x2222222222222222222222222222222222222222222222222222222222222222, 
    },
    ];

    // Accessing an element of an array
    let array_of_bools: [bool;
    2] = [true, false];
    assert(array_of_bools[0]);
}

```
is formatting to:
```rust
script;

use std::assert::assert;

struct Foo {
    f1: u32,
    f2: b256,
}

fn main() {
    // Array of integers with type ascription
    let array_of_integers: [u8; 5] = [1, 2, 3, 4, 5];


    // Array of strings
    let array_of_strings = [
        "Bob",
        "Jan",
        "Ron",
    ];


    // Array of structs
    let array_of_structs: [Foo; 2] = [
        Foo {
            f1: 11,
            f2: 0x1111111111111111111111111111111111111111111111111111111111111111,
        },
        Foo {
            f1: 22,
            f2: 0x2222222222222222222222222222222222222222222222222222222222222222,
        },
    ];


    // Accessing an element of an array
    let array_of_bools: [bool; 2] = [true, false];
    assert(array_of_bools[0]);
}

```
Looks like it's just adding one too many now

**Comment by eureka-cpu** [comment]
Tried this:
```rust
fn format_comment(comment: &Comment) -> String {
    let mut comment_str = comment.span().str();
    if comment_str.starts_with('\n') {
        comment_str.remove(0);
    }
    comment_str
}
```
but it didn't fix it, possibly not the comment itself then?

**Comment by eureka-cpu** [comment]
A `println` of the formatted code before `handle_newlines()` but after `handle_comments()` looks like:
```rust
script;
use std::assert::assert;
struct Foo {
    f1: u32,
    f2: b256,
}
fn main() {
    // Array of integers with type ascription
    let array_of_integers: [u8; 5] = [1, 2, 3, 4, 5];

    // Array of strings
    let array_of_strings = [
        "Bob",
        "Jan",
        "Ron",
    ];

    // Array of structs
    let array_of_structs: [Foo; 2] = [
        Foo {
            f1: 11,
            f2: 0x1111111111111111111111111111111111111111111111111111111111111111,
        },
        Foo {
            f1: 22,
            f2: 0x2222222222222222222222222222222222222222222222222222222222222222,
        },
    ];

    // Accessing an element of an array
    let array_of_bools: [bool; 2] = [true, false];
    assert(array_of_bools[0]);
}
```
so it does seem to be the comments that have the newline

**Comment by eureka-cpu** [comment]
The spans of the comments look fine though... 🤔 
```sh
[swayfmt/src/utils/map/comments.rs:233] comment.span().str() = "// Array of integers with type ascription"
[swayfmt/src/utils/map/comments.rs:233] comment.span().str() = "// Array of strings"
[swayfmt/src/utils/map/comments.rs:233] comment.span().str() = "// Array of structs"
[swayfmt/src/utils/map/comments.rs:233] comment.span().str() = "// Accessing an element of an array"
```

**Comment by eureka-cpu** [comment]
Looks like the problem is happening during `format_context()`, this is the result of the context + comment span:
```sh
"
    // Array of integers with type ascription"
"

    // Array of strings"
"

    // Array of structs"
"

    // Accessing an element of an array"
```

**Comment by eureka-cpu** [comment]
I added a quick fix in #1516 just so that the PR can get an approval but it looks like a more robust method is needed. Instances where you _do_ want newlines between comments end up looking like:
```rust
///////////
// comment block
///////////
// this is another comment why no newline
/// doc comment that was supposed to have newline before it
```
I think part of the problem is the interaction between comment and newline handling still

**Comment by eureka-cpu** [comment]
Closing this in favor of #2790
